components/announcement_bar/configuration_bar/index.ts

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {connect} from 'react-redux';
+import {bindActionCreators, Dispatch} from 'redux';
+
+import {GenericAction} from 'mattermost-redux/types/actions';
+
+import {GlobalState} from 'types/store';
+
+import {AnnouncementBarMessages} from 'utils/constants';
+import {dismissNotice} from 'actions/views/notice';
+import {getSiteURL} from 'utils/url';
+
+import ConfigurationBar from './configuration_bar';
+
+function mapStateToProps(state: GlobalState) {
+    return {
+        siteURL: getSiteURL(),
+        dismissedExpiringLicense: Boolean(state.views.notice[AnnouncementBarMessages.LICENSE_EXPIRING]),
+        dismissedNumberOfActiveUsersWarnMetricStatus: Boolean(state.views.notice[AnnouncementBarMessages.WARN_METRIC_STATUS_NUMBER_OF_USERS]),
+        dismissedNumberOfActiveUsersWarnMetricStatusAck: Boolean(state.views.notice[AnnouncementBarMessages.WARN_METRIC_STATUS_NUMBER_OF_USERS_ACK]),
+        dismissedNumberOfPostsWarnMetricStatus: Boolean(state.views.notice[AnnouncementBarMessages.WARN_METRIC_STATUS_NUMBER_OF_POSTS]),
+        dismissedNumberOfPostsWarnMetricStatusAck: Boolean(state.views.notice[AnnouncementBarMessages.WARN_METRIC_STATUS_NUMBER_OF_POSTS_ACK]),
```

Comment:
As you pointed out, the `hasBeenDismissed` should be added to the `GlobalState` so that it can properly access that map correctly for all of these 
```suggestion
        dismissedNumberOfPostsWarnMetricStatusAck: Boolean(state.views.notice.hasBeenDismissed[AnnouncementBarMessages.WARN_METRIC_STATUS_NUMBER_OF_POSTS_ACK]),
```

Reply:
```typescript
        notice: {
            hasBeenDismissed: {
                [message: string]: boolean;
            };
            [noticeType: string]: boolean;
        };
```

My first inclination was to try something like this. However, this is not valid syntax.

I only see `hasBeenDismissed` used in code, so I've removed the index signature for now.